### ISSUE TITLE
Reuse generated images in train_step

### ISSUE THREAD

**Issue by jamadaha** [jamadaha/P5#153]
**Is your feature request related to a problem? Please describe.**

Currently we generate images twice in train_step, it might be fruitful to reuse them instead.
See here: https://www.tensorflow.org/tutorials/generative/dcgan

**Comment by kris701** [comment]
Looking into this a bit, i dont think we can do it just like in that DCGAN example.
The major issue is that we need to train the discriminator first, before the generator, and in that turn the first generated_images thats made, the discriminator "learns" to perfectly interpret. 
This means that when it comes to training the generator, the gradients from the discriminator is None, since the discriminator already "knows" those images.
The issue is that we also have labels, and not just images as the DCGAN uses.
